### DISCUSSION TITLE
try only block

### DISCUSSION THREAD

**Discussion by sellep** [dotnet/csharplang#2151] [Language Ideas] [answered]
Sometimes I just want to 'try' something without being interested in exception handling

```
try
{
    // something I want todo
}
catch
{

}
```

Which could be shorten to
```
try
{
    // something I want todo, all exceptions are catched but not handled
}

```

**Comment by CyrusNajmabadi** [comment, marked as answer]
Since you can already write this as `catch {}`, this seems like a non-starter proposal.  

1. This is saving almost nothing (just 3 tokens).
2. This sort of behavior is exceedingly rare.
3. This sort of behavior is discouraged as swallowing exceptions is considered a very bad practice.

I cannot envision a world where the language would put in the effort to support something that would be near universally panned as a bad idea.

**Comment by whoisj** [comment]
I think M# had something like `try [statement]` and it an exception was throw the app would fail-fast. We could go that route </sarcasm> :smirk:

**Comment by sellep** [comment]
Consider a public API, which has one or more events:
```
public class Api
{
    public event EventHandler Change;

    public void DoSomething() => RaiseChange();

    private void RaiseChange()
    {
        try { Change?.Invoke(this, new EventArgs()); } catch { }
    }
}
```
When a consumer of this API is not handling possible exceptions, they will break through the API:
```
public class Program
{

    static void Main(string[] args)
    {
        Api api = new Api();
        api.Change += (s, e) => throw new NotImplementedException();
        api.DoSomething();
    }
}
```
Why should the API take care of any of such exceptions?
I wouldn't even log them, since absolutely not the API's responsibility

Actually this was my intention

**Comment by HaloFour** [comment]
> Consider a public API, which has one or more events:

You should let them bubble up.  Ignoring them is certainly not the right thing to do.
